### PR TITLE
[WIP] Vue Support

### PR DESCRIPTION
**Pull request by marceloavf**
### Description

Initial implementation of a plugin that integrates support for Vue componentes into Docz

#### [react-vue-loader](https://github.com/SmallComfort/react-vue-loader)
- Would be the greatest option, since it's just integrate inside webpack configuration, problems:
   - Doesn't support webpack 4: https://github.com/SmallComfort/react-vue-loader/issues/7

#### [vuera](https://github.com/akxcv/vuera)
- Another great option, just need to include babel configuration for the fastest implementation, problems:
  - Docz don't seems to be understand the babel configuration for `vuera/babel`

### Review

- [ ] Code implementation

### Pre-merge checklist

- [ ] Need to build `examples/vue` folder
- [ ] Need to test `<PropsTable>` integration

---

***Warning:** this is a working in progress*

---

### CODE REVIEW

**Comment by pedronauck** [comment]
@marceloavf you need to transpile the plugin, import on `doczrc.js` and use it, like that:

```js
import { vue } from 'docz-plugin-vue'

export default {
  plugins: [vue()],
}
```

**Comment by marceloavf** [comment]
It's now building with some console errors:

```shell
Uncaught TypeError: Cannot read property 'constructor' of undefined
    at isReactComponent (vuera.cjs.js?ea64:410)
    at babelReactResolver$$1 (vuera.cjs.js?ea64:475)
    at ProxyComponent.render (root.jsx?1914:29)
    at ProxyComponent.hotComponentRender (react-hot-loader.development.js?ba54:622)
    at ProxyComponent.proxiedRender (react-hot-loader.development.js?ba54:630)
    at finishClassComponent (react-dom.development.js?6df8:13085)
    at updateClassComponent (react-dom.development.js?6df8:13047)
    at beginWork (react-dom.development.js?6df8:13715)
    at performUnitOfWork (react-dom.development.js?6df8:15741)
    at workLoop (react-dom.development.js?6df8:15780)
```
```shell
The above error occurred in the <Root> component:
    in Root (created by HotExportedRoot)
    in AppContainer (created by HotExportedRoot)
    in HotExportedRoot

React will try to recreate this component tree from scratch using the error boundary you provided, AppContainer.
```

**Comment by marceloavf** [comment]
Seems to be something with `react-hot-loader`, I'll do some deep debug in it and research in community

File: packages/docz-plugin-vue/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+export const doczPluginVue = () =>
+  createPlugin({
+    modifyBabelRc: ((babelrc: BabelRC): BabelRC => {
+      if (babelrc.plugins === undefined) {
+        babelrc.plugins = []
+      }
+      babelrc.plugins.push('vuera/babel')
+      babelrc.babelrc = true
+      return babelrc
+    }) as ModifyBabelRC,
```
Comment (pedronauck) [inline packages/docz-plugin-vue/src/index.ts]: I think that ts will infer this for you... you really need this `as <Something>`?

File: packages/docz-plugin-vue/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    modifyBundlerConfig: (config => {
+      config.module.rules.push({
+        test: /\.vue$/,
```
Comment (pedronauck) [inline packages/docz-plugin-vue/src/index.ts]: Try to put some `exclude` and `include` paths here, I think that you that you can prevent some parses!

**Review by pedronauck — CHANGES_REQUESTED** [review]

**Comment by marceloavf** [comment]
The plugin seems to be compiling with some warning, need to investigate.

**Pull request closed without merging**